### DISCUSSION TITLE
OpenSSL@1.1 is installed, but /etc/ssl/cert.pem is not updated

### DISCUSSION THREAD

**Discussion by trietsch** [Homebrew/discussions#2444] [Everyday usage] [answered]
As the title states, the `/etc/ssl/cert.pem` is not updated when installing OpenSSL@1.1. Many libraries and binaries, such as cURL and librdkafka, use the CA cert file from this location by default.
Is it correct that `/etc/ssl/cert.pem` is not updated upon installation of OpenSSL@1.1?

At the moment, the `/etc/ssl/cert.pem` is outdated on my machine, which breaks connections using Let's Encrypt based certificates which are signed with both the old (and expired) Root CA certificate "DST Root CA X3".

**Comment by Bo98** [comment, marked as answer]
`/etc/ssl/cert.pem` is supplied by Apple and is not controlled or touched by us.

Our OpenSSL and curl does not use it by default. It uses `HOMEBREW_PREFIX/etc/openssl@1.1/cert.pem` (which is a symlink to `HOMEBREW_PREFIX/etc/ca-certificates/cert.pem`).

**Reply by trietsch** [reply]
Ah I didn't know that. Maybe librdkafka should use that directory by default as well. Thanks for the answer. To be honest, Apple should just update the certificates at `/etc/ssl/cert.pem`, but I don't see that happening anytime soon.

**Reply by carlocab** [reply]
> Maybe librdkafka should use that directory by default as well.

Does it not use OpenSSL's CA store? If it does, it should use the `pem` file installed by `ca-certificates`.

**Reply by trietsch** [reply]
I read something about how librdkafka uses the CA file location today [here](https://github.com/edenhill/librdkafka/blob/master/INTRODUCTION.md#ssl). I've posted my question at librdkafka as well: https://github.com/edenhill/librdkafka/discussions/3618

**Comment by carlocab** [comment]
To add to Bo's answer: installing and uninstalling Homebrew formulae never touches anything outside of `HOMEBREW_PREFIX`, and never runs using `sudo`. You definitely need at least `sudo` to modify `/etc/ssl/cert.pem` (if it's not protected by SIP -- it might be).
